flocker/control/_model.py

```diff
@@ -930,6 +950,38 @@ def get_information_wipe(self):
         return _WipeNodeState(node_uuid=self.uuid, attributes=attributes)
 
 
+@implementer(IClusterStateChange)
+class UpdateNodeStateEra(PClass):
+    """
+    Update a node's era.
+
+    :ivar UUID uuid: The node's UUID.
+    :ivar UUID era: The node's era.
+    """
+    uuid = field(type=UUID, mandatory=True)
+    era = field(type=UUID, mandatory=True)
+
+    def update_cluster_state(self, cluster_state):
+        """
+        Record the node's era and discard the ``NodeState`` if it doesn't
+        match the era.
+        """
+        if cluster_state.node_uuid_to_era.get(self.uuid) != self.era:
+            # Discard the NodeState:
+            cluster_state = cluster_state.set(
+                nodes={n for n in cluster_state.nodes
+                       if n.uuid != self.uuid})
+        cluster_state = cluster_state.transform(
+            ["node_uuid_to_era", self.uuid], self.era)
+        return cluster_state
+
+    def get_information_wipe(self):
+        """
+        Since we just deleted some information, there's nothing to wipe.
```

Comment:
Is this accurate? One could imaging a wipe implementation that removed the uuid -> era mapping. I think we're not doing that here because we expect the next incarnation to wipe it out for us instead of using the wiping mechanism here.